src/ts/interfaces.ts

```diff
@@ -19,6 +19,7 @@ interface PartialEntity {
 }
 
 interface Upload {
+  id?: string;
```

Comment:
shouldn't it be a `number`?

Reply:
Yes it should be but the data returned from the request handler is all strings (see below) and casting to the Upload interface does not change it (see image) because at this point it is just JS and not TS. So the proper expression would be `tinymceEditImage.uploadId = parseInt(String(upload.id), 10);` but that seems odd as well: from `string` → `int` → `string` → `int`? So in the end it might be okay to have the id declared as a string?
~~~JSON
{
  "res": true,
  "msg": "Saved",
  "value": {
    "id": "1452",
    "real_name": "Nico.png",
    "long_name": "ef/ef15d9779f13464cb5e0c925e7f000d159163a75d04706077068a9cfd6c2675a6756a301f7ae5cedb902445979553e425cb960c27418da9983431b9408fdb2be.png",
    "comment": "Click to add a comment",
    "item_id": "2600",
    "userid": "1",
    "type": "experiments",
    "created_at": "2022-07-13 21:36:35",
    "hash": "c97bd32fee1e3d8b1026b3c8c1e7c06ac55f1ff7ac8e20b9b08acdb8d2986c9a",
    "hash_algorithm": "sha256",
    "storage": "1",
    "filesize": "11819",
    "state": "1"
  }
}
~~~
![image](https://user-images.githubusercontent.com/65481677/179065371-62107cff-2366-4aa4-9ab2-261eec735f55.png)

Reply:
> Like a type annotation, type assertions are removed by the compiler and won’t affect the runtime behavior of your code.

from https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions
So am I abusing it here?